flair/models/clustering/kmeans/k_Means.py

```diff
@@ -0,0 +1,84 @@
+import copy
+
+import numpy as np
+import torch
+
+from flair.datasets import DataLoader
+from flair.embeddings import DocumentEmbeddings
+from flair.models.clustering.clustering import Clustering
+from flair.models.clustering.distance import distance
+
+
+class KMeans(Clustering):
+    def __init__(self, k, embeddings: DocumentEmbeddings = None):
+        self.k = k
+        self.embeddings = embeddings
+        self.k_points = []
+        self.predict = []
+        self.max_iteration = 100
+
+    def cluster(self, sentences: list, batch_size: int = 64) -> list:
+        print("k Means cluster start with k: " + str(self.k))
+
+        for batch in DataLoader(sentences, batch_size=batch_size):
+            self.embeddings.embed(batch)
+
+        vectors = [sentence.embedding for sentence in sentences]
+        cluster_result = self.cluster_vectors(vectors)
+
+        for idx, sentence in enumerate(sentences):
+            sentence.set_label("cluster", str(self.predict[idx]))
+
+        return cluster_result
+
+    def cluster_vectors(self, vectors: list) -> list:
+        self.predict = [0] * len(vectors)
+
+        for i in range(self.k):
```

Comment:
why do you iterate here k-times and sample k sentences as cluster centers? Once should be enough.

Reply:
and thus you are overriding k_points k-times